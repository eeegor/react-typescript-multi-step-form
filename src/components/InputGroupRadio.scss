.input-radio {
	background-color: initial;
	cursor: default;
	margin: 0;
	padding: 0;
	border: 0;
	font-size: 1rem;
	height: 0.9rem;
	width: 0.9rem;
	text-align: center;
	display: inline-flex;
	position: relative;
	top: 0.1rem;
	margin-right: 0.5rem;
}

.input-group-radio .input-group__label {
	font-size: 1rem;
	font-weight: 400;
}
