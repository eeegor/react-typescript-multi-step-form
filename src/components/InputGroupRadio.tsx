/**
 * Component InputGroupRadio
 */

import * as React from 'react';
import './InputGroupRadio.scss';

export interface Props {
	id: string;
	name: string;
	label: string;
	value: string;
	type?: string;
	info?: string;
	defaultChecked?: boolean;
	onChange: (name: string, event: React.ChangeEvent<HTMLInputElement>) => void;
}

export interface State {}

export class InputGroupRadio extends React.Component<Props, State> {
	constructor(props: Props) {
		super(props);
	}

	render(): JSX.Element {
		const { id, value, onChange, label, name, defaultChecked, info } = this.props;

		return (
			<div className="input-group input-group--radio">
				<input
					className="input-group__input input-radio"
					id={id}
					name={name}
					type="radio"
					onChange={event => onChange(name, event)}
					value={value}
					defaultChecked={defaultChecked || false}
				/>
				<label className="input-group__label" htmlFor={id}>
					{label}
				</label>
				{info && <small className="input-group__info">{info}</small>}
			</div>
		);
	}
}
