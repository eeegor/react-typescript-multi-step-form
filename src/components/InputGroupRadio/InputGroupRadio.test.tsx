import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { InputGroupRadio } from './InputGroupRadio';
import { mount } from 'enzyme';

describe('InputGroupRadio', () => {
	const callback = jest.fn();
	const testData = {
		id: 'name',
		name: 'name',
		label: 'Your Name',
		value: 'Bob',
		onChange: callback,
	};

	it('renders markup correctly', () => {
		const component = <InputGroupRadio {...testData} />;
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('change event works', () => {
		const wrapper = mount(<InputGroupRadio {...testData} />);
		wrapper.find('.input-group__input').simulate('change');
		expect(callback).toHaveBeenCalledTimes(1);
	});

	it('should adopt a valid info attribute', () => {
		const info = 'test';
		const wrapper = mount(<InputGroupRadio info={info} {...testData} />);
		expect(
			wrapper
				.find('.input-group__info')
				.first()
				.text()
		).toEqual(info);
	});
});
