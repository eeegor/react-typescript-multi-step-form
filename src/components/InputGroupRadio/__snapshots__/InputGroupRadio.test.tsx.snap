// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroupRadio renders markup correctly 1`] = `
<div
  className="input-group input-group-radio"
  id="input-group-name"
>
  <input
    autoFocus={false}
    className="input-group__input input-radio"
    defaultChecked={false}
    id="name"
    name="name"
    onChange={[Function]}
    type="radio"
    value="Bob"
  />
  <label
    className="label input-group__label"
    htmlFor="name"
  >
    Your Name
  </label>
</div>
`;
